c/secp256k1_blake160_sighash_all.c

```diff
@@ -186,8 +163,25 @@ int main(int argc, char* argv[])
     }
     blake2b_final(&blake2b_ctx, temp, BLAKE2B_BLOCK_SIZE);
 
-    if (secp256k1_ecdsa_verify(&context, &signature, temp, &pubkey) != 1) {
-      return ERROR_SECP_VERIFICATION;
+    secp256k1_pubkey pubkey;
+
+    if (secp256k1_ecdsa_recover(&context, &pubkey, &signature, temp) != 1) {
```

Comment:
I'm confused here, I only saw for for parsing signature, extracting pubkey, then does pubkey hash check. Where's the actual signature verification code?

Reply:
Because of the pubkey is recovered from the signature, it implies the signature is signed by the seckey which derive the pubkey, this step already does the verify, we do not need to verify the signature again. So the only thing needs to verify is to check the hash of recovered pubkey.